starship/src/configs/rust.rs

```diff
@@ -0,0 +1,73 @@
+use crate::module_config::ModuleConfig;
+
+use ansi_term::{Color, Style};
+use starship_module_config_derive::ModuleConfig;
+
+#[derive(Clone, ModuleConfig, Debug)]
+pub struct RustConfig<'a> {
+    pub symbol: &'a str,
+    pub style: Style,
+    pub disabled: bool,
+}
+
+/* This works
+impl<'a> ModuleConfig<'a> for RustConfig<'a> {
+    fn load_config(&self, config: &'a toml::Value) -> Self {
+        let mut new_module_config = self.clone();
+        if let toml::Value::Table(config) = config {
+            if let Some(config_str) = config.get("symbol") {
+                if let Some(symbol) = <&str>::from_config(config_str) {
+                    new_module_config.symbol = symbol;
+                }
+            }
+            if let Some(config_str) = config.get("disabled") {
+                if let Some(disabled) = <bool>::from_config(config_str) {
+                    new_module_config.symbol = disabled;
+                }
+            }
+            if let Some(config_str) = config.get("style") {
+                if let Some(style) = <Style>::from_config(config_str) {
+                    new_module_config.symbol = style;
+                }
+            }
+        }
+        new_module_config
+    }
+}
+*/
```

Comment:
This is what `#[derive(ModuleConfig)]` will generate for you.

TODO: delete the comment before merging.

Reply:
Do I understand it correctly that it will fail to load the config if any of the required properties are missing?

I really love the idea of automatic config loader derivation but I'm worried that it doesnt cover default values. So either users will have to set all properties even though they might only want to customize one or every property is marked optional with unwrap_or across the module implementations

Reply:
No, `load_config` will merge user's config with `Self`, which is probably the default config, while `from_config` will fail if any of the items is missing.

Reply:
You can refer to [this test](https://github.com/starship/starship/blob/dd98c7222358957e15ddc6c1d55ceec4fd32ebc7/starship/src/config.rs#L385)

Reply:
Uhh. I didn't see that. That's amazing!